### PR TITLE
Partially remove dependency on imp.

### PR DESCRIPTION
**Pull request by degustaf**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [ ] Write a good description on what the PR does.

## Description

This partially removes the dependency on the deprecated imp module.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

## Related Issue
This is related to issues #594 and #681

### CODE REVIEW

**Comment by degustaf** [comment]
Note that removing the final dependency on imp requires making a decision. It is located in `modutils.load_module_from_modpath` and is a pair of calls to `imp.find_module` and `imp.load_module`. The official documentation recommends replacing these with `importlib.import_module`. For this use case it does not allow for ignoring `sys.modules`, which the API allows. 

I see 2 options for how to solve this:

1. Change the API to remove the option to ignore `sys.module`. This would affect the functions `modutils.load_module_from_modpath`, `modutils.load_module_from_name`, and `modutils.load_module_from_file`.  This option does not appear to be used anywhere in astroid or pylint, except for 1 unittest.

2. Copy the code from imp.py so that we maintain the functionality by maintaining the code.

File: astroid/interpreter/_import/spec.py [hunk 1]
```diff
@@ -116,23 +96,62 @@ class ImpFinder(Finder):
...
-        if stream:
-            stream.close()
-
-        return ModuleSpec(
-            name=modname,
-            location=mp_filename,
-            module_type=_imp_type_to_module_type(mp_desc[2]),
+        else:
+            try:
+                loader = importlib.util.find_spec(modname)
```
Comment (gyermolenko) [inline astroid/interpreter/_import/spec.py]: Just curious. The function returns `ModuleSpec` object, why did you name it `loader`?

**Comment by gyermolenko** [comment]
Don't know if it would help, but here is ref to similar improvement in pytest-dev/pytest#5468

**Comment by PCManticore** [comment]
This is on my todo list for reviewing. Unfortunately might take me a bit as I'll be in vacation for the next two weeks.

File: astroid/interpreter/_import/spec.py [hunk 2]
```diff
@@ -116,23 +96,62 @@ class ImpFinder(Finder):
...
+                        return ModuleSpec(
+                            name=modname,
+                            location=None,
+                            module_type=ModuleType.PY_FROZEN,
+                        )
+            except ValueError:
+                pass
+            submodule_path = sys.path
+
+        suffixes = (
```
Comment (PCManticore) [inline astroid/interpreter/_import/spec.py]: Is it possible to build this list globally instead of building it every time this function is called?

**Comment by PCManticore** [comment]
@degustaf Can you rebase your PR when you get a chance? I'd like to pull this in. Thank you!

**Comment by degustaf** [comment]
@PCManticore I should be able to do that tonight.

**Comment by stonebig** [comment]
any hope on this ?

**Comment by hippo91** [comment]
I close this issue because #857 has been merged.

**Pull request closed without merging**